fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java

```diff
@@ -1944,12 +1987,13 @@ public FragmentExecParams(PlanFragment fragment) {
             }
 
             List<TExecPlanFragmentParams> paramsList = Lists.newArrayList();
-            for (int i = 0; i < instanceExecParams.size(); ++i) {
-                final FInstanceExecParam instanceExecParam = instanceExecParams.get(i);
+            int startIdx = backendNum;
+            for (int i = 0; i < fInstanceExecParamList.size(); ++i) {
+                final FInstanceExecParam instanceExecParam = fInstanceExecParamList.get(i);
                 TExecPlanFragmentParams params = new TExecPlanFragmentParams();
 
                 if (exportSink != null && fileNamePrefix != null) {
-                    exportSink.setFileNamePrefix(fileNamePrefix + i + "_");
+                    exportSink.setFileNamePrefix(fileNamePrefix + (startIdx + i) + "_");
```

Comment:
why change this?

Reply:
in order to keep consistent with the original logic. In old implementation, toThrift is invoked only once for a PlanFragment, so for each FragmentInstance, here i increasing from zero is ok. but when desc_tbl cache is introduced, for a PlanFragment, toThrift is invoked twice, for the 2nd invocation, here i should increase from the base of 1st invocation.